lib/tpm2_nv_util.h

```diff
@@ -48,12 +48,10 @@
  *  True on success, false otherwise.
  */
 static inline bool tpm2_util_nv_read_public(TSS2_SYS_CONTEXT *sapi_context,
-        TPMI_RH_NV_INDEX nv_index, TPM2B_NV_PUBLIC *nv_public) {
-
-    TPM2B_NAME nv_name = TPM2B_TYPE_INIT(TPM2B_NAME, name);
+        TPMI_RH_NV_INDEX nv_index, TPM2B_NV_PUBLIC *nv_public, TPM2B_NAME *nv_name) {
```

Comment:
Document this new parameter. Can you allow NULL here? This way the nvread/write callers can just specify NULL and not have to add a variable that is never really used?
```c
TPM2B_NAME _nv_name = TPM2B_TYPE_INIT(TPM2B_NAME, name);
if (!nv_name) {
   nv_name = &_nv_name;
}

TSS2_RC rval = TSS2_RETRY_EXP(Tss2_Sys_NV_ReadPublic(sapi_context, nv_index, NULL, nv_public,
            nv_name, NULL));
...
```